### PR TITLE
Conditionally enable the HostTelemetry process

### PR DESCRIPTION
**Pull request by fmterrorf**
Add METRIST_ENABLE_HOST_TELEMETRY to conditionally enable the HostTelemetry process

**Deployment process**
See related PR

### CODE REVIEW

File: lib/orchestrator/application.ex [hunk 1]
```diff
@@ -104,30 +101,13 @@ defmodule Orchestrator.Application do
...
-        token ->
-          Orchestrator.Configuration.translate_value(token)
-      end
-
-    Application.put_env(:orchestrator, :api_token, token)
-  end
-  defp configure_slack_reporter do
-    token = System.get_env("SLACK_API_TOKEN")
-    |> Orchestrator.Configuration.translate_value()
-    Application.put_env(:slack, :api_token, token)
```
Comment (fmterrorf) [inline lib/orchestrator/application.ex:126]: There is no `:slack` application (and Mix config complains if you add a `config :slack` app) so I moved this to `slack_api_token` under the orchestrator configs.

File: config/runtime.exs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import Config
+
+if config_env() == :prod do
+  config :orchestrator,
+    enable_host_telemetry?:
+      System.get_env("METRIST_ENABLE_HOST_TELEMETRY", "true") |> String.to_existing_atom(),
```
Comment (cdegroot) [inline config/runtime.exs:6]: I think we want the default to be "false" here.

File: lib/orchestrator/application.ex [hunk 3]
```diff
@@ -104,30 +101,13 @@ defmodule Orchestrator.Application do
...
-
-    channel = System.get_env("SLACK_ALERTING_CHANNEL")
-    |> Orchestrator.Configuration.translate_value()
-    Application.put_env(:slack, :reporting_channel, channel)
+  defp filter_children(children) do
```
Comment (cdegroot) [inline lib/orchestrator/application.ex:105]: Isn't this a bit "dirty" versus the other way around, only adding the module if we need it?

Comment (fmterrorf) [reply lib/orchestrator/application.ex]: Added  a commit to only add the HostTelemetry module if it's enabled in this commit https://github.com/Metrist-Software/orchestrator/pull/114/commits/a6aa4b25eba46ba8e1ad07045a16634a9f67d17a

**Review by cdegroot — APPROVED** [review]

**Pull request merged**